support/bootstrap

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+#
+# Bootstraps the developer tools and development environment. This includes
+# copying project-standard git commit hooks that do things like ensure DCO
+# signoff is present during commit.
+
+# Best-effort check that we're in the envoy root directory.
+#
+# TODO(hausdorff): If compatibility becomes an issue here, come back and do this
+# "the right way". This is hard to do in general, since `realpath` is not
+# standard.
+if [ ! "$PWD" == "$(git rev-parse --show-toplevel)" ]; then
+    cat >&2 <<__EOF__
+ERROR: this script must be run at the root of the envoy source tree
```

Comment:
This line might need to be modified.

Also keep in mind that folks using nighthawk are likely Envoy developers and would already have the bootstrap scripts in stalled from there.

Ultimately I'm not too keen about adding the changes in this PR.

You could also refer to the DEVELOPER.md in the Envoy repo to refer to the bootstrap setup.  Not certain this project needs its own scripts.